### PR TITLE
Add snapcraft build support

### PR DESCRIPTION
**Pull request by MrCarroll**
This commit adds Snapcraft support. To use it, build the launcher with Maven first and then run the snapcraft command afterwards.

### CODE REVIEW

**Comment by deathbeam** [comment]
Maybe this PR should include also shell script to build the snap (in similar way what we do atm with other distributions) if that makes sense (i wasnt following the discussion in discord at all btw)

**Comment by MrCarroll** [comment]
There's not a real need for an actual build script, assuming Snapd already exists, the process goes:
sudo snap install snapcraft -- classic (one time only, to get the Snapcraft system)
mvn clean package
snapcraft

If you're running in a VM and your CPU doesn't support nested virtualisation, you might need to run Snapcraft with a different backend, E.G
https://snapcraft.io/docs/build-on-lxd

But that too is a onetime thing, and just adds one parameter to the snapcraft command

So if there was to be a build_snap.sh, it would legitimately just be mvn clean package && snapcraft - actually getting mvn and snapcraft is the users own concern.

File: snap/gui/runelite.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=RuneLite
+Comment=A popular free, open-source and super fast client for Old School Runescape
```
Comment (Adam-) [inline snap/gui/runelite.desktop]:
```suggestion
Comment=A popular free, open-source and super fast client for Old School RuneScape
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+name: runelite
+base: core18
+title: RuneLite
+version: git
+summary: A popular free, open-source and super fast client for Old School Runescape
```
Comment (Adam-) [inline snap/snapcraft.yaml]:
```suggestion
summary: A popular free, open-source and super fast client for Old School RuneScape
```

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+description: |
+  A popular free, open-source and super fast client for Old School Runescape
```
Comment (Adam-) [inline snap/snapcraft.yaml]:
```suggestion
  A popular free, open-source and super fast client for Old School RuneScape
```

**Pull request merged**